engines/pink/objects/actors/lead_actor.cpp

```diff
@@ -0,0 +1,524 @@
+	_state = (State)archive.readByte();
+	_nextState = (State)archive.readByte();
+	_stateBeforeInventory = (State)archive.readByte();
+	_stateBeforePDA = (State)archive.readByte();
+	_isHaveItem = archive.readByte();
+	Common::String recepient = archive.readString();
+	if (!recepient.empty())
+		_recipient = _page->findActor(recepient);
+	else
+		_recipient = nullptr;
+	_sequencer->loadState(archive);
+	_walkMgr->loadState(archive);
+	_page->getGame()->getPdaMgr().loadState(archive);
+	_audioInfoMgr.loadState(archive);
+}
+
+void LeadActor::saveState(Archive &archive) {
+	archive.writeByte(_state);
+	archive.writeByte(_nextState);
+	archive.writeByte(_stateBeforeInventory);
+	archive.writeByte(_stateBeforePDA);
+	archive.writeByte(_isHaveItem);
+	if (_recipient)
+		archive.writeString(_recipient->getName());
+	else
+		archive.writeString(Common::String());
+	_sequencer->saveState(archive);
+	_walkMgr->saveState(archive);
+	_page->getGame()->getPdaMgr().saveState(archive);
+	_audioInfoMgr.saveState(archive);
+}
+
+void LeadActor::init(bool unk) {
+	if (_state == kUndefined)
+		_state = kReady;
+
+	getInventoryMgr()->setLeadActor(this);
+	_page->getGame()->setLeadActor(this);
+	Actor::init(unk);
+}
+
+void LeadActor::start(bool isHandler) {
+	if (isHandler && _state != kPlayingExitSequence) {
+		_state = kPlayingSequence;
+		_nextState = kReady;
+	}
+
+	switch (_state) {
+	case kInventory:
+		startInventory(1);
+		break;
+	case kPDA:
+		if (_stateBeforePDA == kInventory)
+			startInventory(1);
+		loadPDA(_page->getGame()->getPdaMgr().getSavedPageName());
+		break;
+	default:
+		forceUpdateCursor();
+	}
+}
+
+void LeadActor::update() {
+	switch (_state) {
+	case kReady:
+		_sequencer->update();
+		_cursorMgr->update();
+		break;
+	case kMoving:
+		_walkMgr->update();
+		_cursorMgr->update();
+		break;
+	case kPlayingSequence:
+		_sequencer->update();
+		if (!_sequencer->isPlaying()) {
+			_state = _nextState;
+			_nextState = kUndefined;
+			forceUpdateCursor();
+		}
+		break;
+	case kInventory:
+		getInventoryMgr()->update();
+		break;
+	case kPDA:
+		getPage()->getGame()->getPdaMgr().update();
+		break;
+	case kPlayingExitSequence:
+		_sequencer->update();
+		if (!_sequencer->isPlaying()) {
+			_state = kUndefined;
+			_page->getGame()->changeScene();
+		}
+		break;
+	default:
+		break;
+	}
+}
+
+void LeadActor::loadPDA(const Common::String &pageName) {
+	if (_state != kPDA) {
+		if (_state == kMoving)
+			cancelInteraction();
+		if (_state != kInventory)
+			_page->pause(true);
+
+		_stateBeforePDA = _state;
+		_state = kPDA;
+	}
+	_page->getGame()->getDirector()->saveStage();
+	_page->getGame()->getPdaMgr().setLead(this);
+	_page->getGame()->getPdaMgr().goToPage(pageName);
+}
+
+void LeadActor::onKeyboardButtonClick(Common::KeyCode code) {
+	if (code == Common::KEYCODE_g) {
+		loadPDA("TOC");
+		return;
+	}
+	switch (_state) {
+	case kMoving:
+		switch (code) {
+		case Common::KEYCODE_ESCAPE:
+			cancelInteraction();
+			// Fall Through intended
```

Comment:
Please change this to `// fall through` to be consistent with the rest of the tree.